File: lab_5/lab_5_py/lab_5_py/TPyramid.py
from TFigure import *

class TPyramid(TFigure):
    
    __apopheme = 0.0
    __base_side = 0.0

    def __init__(self, pyramid):
        self._height = float(pyramid[0])
        self.__apopheme = float(pyramid[1])
        self.__base_side = float(pyramid[2])

    def find_surface_area(self):
        side_surface_area = 0.5 * (4 * self.__base_side) * self.__apopheme
        base_area = pow(self.__base_side, 2)
        return side_surface_area + base_area

    def find_volume(self):
        return (1/3.0) * pow(self.__base_side, 2) * self._height

    def check_data(self):
        if pow(self.__apopheme, 2) == pow(self._height, 2) + pow(self.__base_side/2.0, 2):
            return True
        else:
            return False

File: lab_5/lab_5_cs/lab_5_cs/TPyramid.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace lab_5_cs
{
    class TPyramid : TFigure
    {
        private double _height;
        private double _apopheme;
        private double _baseSide;

        public double Height {set { _height = value; } }
        public double Apopheme {set { _apopheme = value; } }
        public double BaseSide {set { _baseSide = value; } }

        public override double FindSurfaceArea()
        {
            double sideSurfaceArea = 0.5 * (4 * _baseSide) * _apopheme;
            double baseArea = Math.Pow(_baseSide, 2);
            return sideSurfaceArea + baseArea;
        }

        public override double FindVolume()
        {
            return (1 / 3.0) * Math.Pow(_baseSide, 2) * _height;
        }

        public bool CheckData()
        {
            if (Math.Pow(_apopheme, 2) == Math.Pow(_height, 2) + Math.Pow(_baseSide/2.0, 2))
            {
                return true;
            }
            else { return false; }
        }
    }
}
